modules/DNN/NeuronLayer.cpp

```diff
@@ -4,36 +4,50 @@
 #include "macroses.h"
 
 namespace DirectGraphicalModels { namespace dnn
-{
-	void CNeuronLayer::generateRandomWeights(void)
 	{
-		m_weights = random::U(m_weights.size(), m_weights.type(), -0.5f, 0.5f);
-		m_biases = random::U(m_biases.size(), m_biases.type(), -0.5f, 0.5f);
-	}
+		void CNeuronLayer::generateRandomWeights(void)
+		{
+			m_weights = random::U(m_weights.size(), m_weights.type(), -0.5f, 0.5f);
+			m_biases = random::U(m_biases.size(), m_biases.type(), -0.5f, 0.5f);
+		}
 
-	void CNeuronLayer::dotProd(const Mat& values)
-	{
-		// this->m_netValues = this->m_weights * values + m_biases;
-		gemm(m_weights.t(), values, 1, m_biases, 1, m_netValues);
-	}
+		void CNeuronLayer::dotProd(const Mat& values)
```

Comment:
So the `dotProd(values)` method is: 
_m_netValues = m_weights * **values** + m_biases;_

And the dotProdVis(values, weights) method is: 
_m_netValues = **weights** * **values** + m_biases;_

Reply:
Correct, reason is `dotProd(values)` multiplied the values by this->weights, but `dotProdVis(values, weights)` multiplies values by weights that are passed through. When algorithm updates the visible units, we need to multiply the hidden units by weights, and since we only generate random weights for hidden layer, and we know that _visible units_ = **_hidden weights_** x **_hidden neurons_** , It was not possible to replicate in already available `dotProd(values)`.